src/linker/Linker.Dataflow/ReflectionMethodBodyScanner.cs

```diff
@@ -2151,6 +2166,35 @@ void MarkEventsOnTypeHierarchy (ref ReflectionPatternContext reflectionContext,
 				MarkEvent (ref reflectionContext, @event);
 		}
 
+		void ValidateGenericMethodInstantiation (
+			ref ReflectionPatternContext reflectionContext,
+			MethodDefinition genericMethod,
+			ValueNode typeArgumentsValue,
+			MethodReference reflectionMethod)
+		{
+			if (!genericMethod.HasGenericParameters)
+				return;
+
+			// If there are no type arguments there's nothing to validate.
+			// This case will typically fail at runtime
+			// since the call requires that all generic parameters have arguments to them.
+			// But it's not the job of the linker to detect possible runtime error cases
+			// the code may well expect to get an exception in this case.
+			if ((typeArgumentsValue as ArrayValue)?.Size.AsConstInt () == 0)
```

Comment:
Will this be `false` if `methodParams[1]` is not `ArrayValue`?

Reply:
The left side should evaluate to `null` and `null != 0`.

Reply:
We could probably delete this for the same reasons as https://github.com/mono/linker/pull/1930#discussion_r606120623.

This would fail at runtime if this is non-null and doesn't match the number of required arguments in general, not just for zero. It feels low value compared to the amount of necessary comments and code.

Reply:
I guess this makes sense - since it will never really work at runtime. Removed.